src/com/mendix/widget/ProgressBar/components/ProgressBar.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, DOM, createElement } from "react";
+import * as classNames from "classnames";
+
+import { Alert } from "./Alert";
+
+interface ProgressBarProps {
+    barType?: BarType;
+    bootstrapStyle?: BootstrapStyle;
+    colorSwitch?: number;
+    contextObjectGuid?: string;
+    maximumValue: number;
+    onClickMicroflow?: string;
+    progress: number | null;
+}
+
+type BootstrapStyle = "default" | "info" | "primary" | "success" | "warning" | "danger";
+type BarType = "default" | "striped" | "animated";
+
+class ProgressBar extends Component<ProgressBarProps, { alertMessage: string }> {
+    static defaultProps: ProgressBarProps = {
+        barType: "default",
+        bootstrapStyle: "default",
+        colorSwitch: 50,
+        maximumValue: 100,
+        progress: 0
+    };
+
+    constructor(props: ProgressBarProps) {
+        super(props);
+
+        this.state = { alertMessage: "" };
+    }
+
+    render() {
+        const { barType, bootstrapStyle, colorSwitch, maximumValue, onClickMicroflow, progress } = this.props;
+        const percentage = this.progressValue(progress, maximumValue);
+        return DOM.div({ className: "widget-progressbar" },
+            DOM.div(
+                {
+                    className: classNames("progress", {
+                        "red-progressbar-text": maximumValue < 1,
+                        "widget-progressbar-clickable": !!onClickMicroflow,
+                        "widget-progressbar-text-contrast": percentage < colorSwitch
+                    }),
+                    onClick: () => this.executeMicroflow(onClickMicroflow, this.props.contextObjectGuid)
+                },
+                DOM.div(
+                    {
+                        className: classNames("progress-bar", `progress-bar-${bootstrapStyle}`, {
+                            "active": barType === "animated",
+                            "progress-bar-striped": barType === "striped" || barType === "animated",
+                            "widget-progressbar-negative": percentage < 0
+                        }),
+                        style: { width: `${Math.abs(percentage)}%` }
+                    },
+                    progress
+                        ? maximumValue < 1 ? "Invalid" : `${this.calculatePercentage(progress, maximumValue)}%`
+                        : ""
+                )
+            ),
+            this.state.alertMessage ? createElement(Alert, { message: this.state.alertMessage }) : null
```

Comment:
Not without having to deal with type errors given the recent strictness. They don't like components returning null. Frankly, I find it better & cleaner this way :)